### PR TITLE
Issue #523 AudioMoth 1.5-1.8.1 MetaData

### PR DESCRIPTION
**Pull request by syunkova**
- Added support for getting AudioMoth metadata incl. comment field for recordings made with firmware versions 1.5-1.8.1
- Fixed typos in comments

### CODE REVIEW

**Review by sammlapp — CHANGES_REQUESTED** [review]
please add tests for each firmware version in test_audiomoth.py

File: opensoundscape/audiomoth.py [hunk 1]
```diff
@@ -159,13 +163,17 @@ def _parse_audiomoth_battery_info(comment):
...
     ...battery state was 3.5V and temperature....
+    ...battery was 4.6V and...
 
     Args:
         comment: the full comment string from an audiomoth metadata Comment field
     Returns:
         float of voltage or string describing voltage, eg "less than 2.5V"
     """
-    battery_str = comment.split("battery state was ")[1].split("V")[0] + "V"
+    if "state " in comment:
```
Comment (jatinkhilnani) [inline opensoundscape/audiomoth.py:173]: Mentioning one more thing discussed earlier. Would be good to change `if "state "` to `if "battery state"` to make it explicit and void changes if another "state" pops up in comments.

File: opensoundscape/audiomoth.py [hunk 2]
```diff
@@ -83,11 +85,13 @@ def parse_audiomoth_metadata(metadata):
...
-        metadata["gain_setting"] = comment.split(" gain setting")[0].split(" ")[-1]
-    # written "3.2V" or "less than 2.5V" (or? greater than 4.5V?)
+    if "gain setting" in comment:
+        try:
+            metadata["gain_setting"] = int(comment.split("gain setting ")[1][:1])
+        except ValueError:
+            metadata["gain_setting"] = comment.split(" gain setting")[0].split(" ")[-1]
+    else:
+        metadata["gain_setting"] = comment.split(" gain")[0].split(" ")[-1]
     metadata["battery_state"] = _parse_audiomoth_battery_info(comment)
```
Comment (jatinkhilnani) [inline opensoundscape/audiomoth.py:95]: Just an empty space before to segregate attribute code block of readability.

**Review by jatinkhilnani — COMMENTED** [review]
Mentioning one more thing discussed earlier. Would be good to change `if "state "` to `if "battery state"` to make it explicit and void changes if another "state" pops up in comments.

File: opensoundscape/audiomoth.py (hunk 2 above)
Comment (sammlapp) [inline opensoundscape/audiomoth.py:95]:
```suggestion

    metadata["battery_state"] = _parse_audiomoth_battery_info(comment)
```

File: opensoundscape/audiomoth.py (hunk 1 above)
Comment (sammlapp) [inline opensoundscape/audiomoth.py:173]:
```suggestion
    if "battery state" in comment:
```

File: tests/test_audiomoth.py [hunk 3]
```diff
@@ -144,3 +144,31 @@ def test_parse_audiomoth_metadata_with_low_battery():
 def test_parse_audiomoth_from_path_wrong_metadata(veryshort_wav_str):
     with pytest.raises(ValueError):
         audiomoth.parse_audiomoth_metadata_from_path(veryshort_wav_str)
+
+
+def test_parse_audiomoth_metadata_v16_and_later():
```
Comment (sammlapp) [inline tests/test_audiomoth.py:149]:
```suggestion
def test_parse_audiomoth_metadata_v16_to_v181():
```

**Review by sammlapp — APPROVED** [review]

**Pull request merged**